linera-core/src/client.rs

```diff
@@ -1137,17 +1239,21 @@ where
     }
 
     /// Updates the latest block and next block height and round information from the chain info.
-    fn update_from_info(&mut self, info: &ChainInfo) {
-        if info.chain_id == self.chain_id && info.next_block_height > self.next_block_height {
-            self.next_block_height = info.next_block_height;
-            self.block_hash = info.block_hash;
-            self.timestamp = info.timestamp;
+    fn update_from_info(&self, info: &ChainInfo) {
+        // TODO (b) this deadlocks with (a)
```

Comment:
TODOs must have task numbers. Also, this one I don't understand the text.

Reply:
This is a stale comment, my bad!